packages/cojson/src/coValueCore/coValueCore.ts

```diff
@@ -208,43 +216,34 @@ export class CoValueCore {
 
   waitForAvailableOrUnavailable(): Promise<CoValueCore> {
     return new Promise<CoValueCore>((resolve) => {
-      const listener = (core: CoValueCore) => {
+      this.subscribe((core, unsub) => {
```

Comment:
replacing manual adding/removing of subscriptions with just calling this.subscribe to have less duplicated logic (now that adding/removing subs is more complex)